source/common/upstream/cds_api_delta_impl.cc

```diff
@@ -0,0 +1,83 @@
+#include "common/upstream/cds_api_delta_impl.h"
+
+#include <string>
+
+#include "envoy/api/v2/cds.pb.validate.h"
+#include "envoy/api/v2/cluster/outlier_detection.pb.validate.h"
+#include "envoy/stats/scope.h"
+
+#include "common/common/cleanup.h"
+#include "common/config/resources.h"
+#include "common/config/subscription_factory.h"
+#include "common/config/utility.h"
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+CdsApiPtr CdsApiDeltaImpl::create(const envoy::api::v2::core::ConfigSource& cds_config,
+                                  ClusterManager& cm, Event::Dispatcher& dispatcher,
+                                  Runtime::RandomGenerator& random,
+                                  const LocalInfo::LocalInfo& local_info, Stats::Scope& scope,
+                                  Api::Api& api) {
+  return CdsApiPtr{new CdsApiDeltaImpl(cds_config, cm, dispatcher, random, local_info, scope, api)};
+}
+
+CdsApiDeltaImpl::CdsApiDeltaImpl(const envoy::api::v2::core::ConfigSource& cds_config,
+                                 ClusterManager& cm, Event::Dispatcher& dispatcher,
+                                 Runtime::RandomGenerator& random,
+                                 const LocalInfo::LocalInfo& local_info, Stats::Scope& scope,
+                                 Api::Api& api)
+    : cm_(cm), scope_(scope.createScope("cluster_manager.cds.")) {
+  Config::Utility::checkLocalInfo("cds", local_info);
+  subscription_ =
+      Config::SubscriptionFactory::subscriptionFromConfigSource<envoy::api::v2::Cluster>(
+          cds_config, local_info, dispatcher, cm, random, *scope_, "not implemented",
+          "envoy.api.v2.ClusterDiscoveryService.DeltaClusters", api);
+}
+
+void CdsApiDeltaImpl::onConfigUpdate(
```

Comment:
I think if we can do the full-update-in-terms-of-incremental thing, we won't need to fork this file, only change the type signature of `onConfigUpdate`. This then lowers the barrier-to-entry for other xDS to use delta updates.

Reply:
Ahhh, now I see. Just one CdsApiImpl, with both types of onConfigUpdate()s implemented. That makes sense; much cleaner!